torchao/quantization/pt2e/_affine_quantization.py

```diff
@@ -113,7 +113,8 @@ def _get_reduction_params(block_size, input_size):
           shape_for_reduction: (3, 3, 5, 2, 10)
           reduction_dim: [0, 1, 3, 4]
     """
-    assert len(block_size) == len(input_size)
```

Comment:
is this still used? we should be using the code in quant_primitives.py I think

Reply:
Yes. It's still used when running the prepared model (model after `prepare_pt2e`). Is it a bug? Do I need to fix it, too?

Reply:
I am using the observers defined here: torchao/quantization/pt2e/_affine_quantization.py

Reply:
I think we should be using the ones in torchao/quantization/observer.py eventually

only occurrence seems to be https://github.com/pytorch/ao/blob/554cb60c750e6ef31bbcafec74bb76a4578902da/test/quantization/pt2e/test_quantize_pt2e.py#L2531 and we want to update it I think

so if you are adding new things I'd recommend use the one from torchao.quantization

Reply:
Hi @jerryzh168 For the issue I mentioned, do you plan to work on it or do you want us to fix it? Thanks.

Reply:
Hi @jerryzh168 Do you have any suggestions on this? Thanks.

Reply:
I thought we already use the one from torchao: https://github.com/pytorch/ao/blob/212d912dcbfb43b13cc4bee533d3d1844d76b7c1/torchao/quantization/pt2e/prepare.py#L35 but if you saw we are using torch.ao please go ahead and change them

Reply:
@jerryzh168 Thanks for the reply. I did not mean `torch.ao`. I meant there are two versions of such utilities in torchao, `torchao.quantization.pt2e` and `torchao.quantization`. For example,
https://github.com/pytorch/ao/blob/5153bd3ce9fc4e873a00d7a24000114ce93a2899/torchao/quantization/pt2e/observer.py#L84
and
https://github.com/pytorch/ao/blob/5153bd3ce9fc4e873a00d7a24000114ce93a2899/torchao/quantization/observer.py#L32
The PT2E flow in torchao uses those in `torchao.quantization.pt2e` while you said you wanted to switch to `torchao/quantization/observer.py`.
So, I was asking whether you would switch to `torchao/quantization/observer.py` in PT2E flow first. Do you have any suggestions on that? Thanks.

Reply:
oh I see, yeah for now use `torchao/quantization/observer.py` would be better I think, we haven't finalized the folder structure for this one yet

Reply:
@jerryzh168 Am I supposed to wait until you finalize the folder structure? Thanks.